### ISSUE TITLE
Camera feed crashes on wye app when server is up

### ISSUE THREAD

**Issue by KhantheDalek** [mrlt8/docker-wyze-bridge#656]
I have just installed DWB on my raspberry pi 4 running homebridge. I have noticed when I have the bridge running and I open the wyze app to view the streams the app can't establish a connection to the cameras.

**Comment by mrlt8** [comment]
Hmm, do you get an error in the app? Are your cameras on the latest firmware? Does changing the stream quality (SD/HD icon) in the app help?

**Comment by jfarmer08** [comment]
I have the same issue. All devices go offline whenever running. It also causes my wyze homebridge plugin to return undefined for all calls. I can try and pull logs at a later date. Just to note after stopping the container alls devices come back online.

**Comment by jfarmer08** [comment]
I enable the setting. And so far yes the devices are still on online. I have one rtsp stream going to homebridge and it seems to be working ok. 

Now I do have 13 cameras and about 84 other wyze devices. No they are not all on the same wifi network. So the on-demand flag can you explain more of what's it's doing? I will let the system run for while and see what happens.

**Comment by mrlt8** [comment]
By default, the bridge will try to connect to all the cameras on the account so that the streams are always available. 

When `on_demand` mode is enabled, the bridge will only connect to the camera when someone requests one of the streams and then disconnects when the stream is closed.

**Comment by jfarmer08** [comment]
Thank you for the answer. This seems to help a lot and I thought my network was doing good.  Are there any recommendations on this subject like number of cameras or setup? The goal would be to use all cameras live stream for a display.

**Comment by jfarmer08** [comment]
So the limit i hit is around 3 to 4. With 3 everthing comes online and goes offline, but with 4 everything goes offline.

**Comment by mrlt8** [comment]
Hmm seems like your network is getting choked with the UDP traffic. 

Unifi router/network by any chance? Those seem to really struggle with the wyze cameras for some reason.

**Comment by talormanda** [comment]
> 

I just installed a ubiquiti and my network loads instantly on the wyze app for my 20+ cameras, not sure if this is necessarily true.

**Comment by talormanda** [comment]
> Hmm seems like your network is getting choked with the UDP traffic.
> 
> Unifi router/network by any chance? Those seem to really struggle with the wyze cameras for some reason.


I just installed a ubiquiti and my network loads instantly on the wyze app for my 20+ cameras, not sure if this is necessarily true. My old setup with tp-link choked under load.

**Comment by ScottySTL** [comment]
The ON_DEMAND setting helped me as well (and I use Unifi networking equipment)

**Comment by James62370** [comment]
My system was choking as well, the ON_DEMAND setting stablized all of my wyze devices.

**Comment by csdozier** [comment]
I still have the issue with the ON_DEMAND setting as well.  If I do 2-3 cameras, its fine, but if I go beyond that, everything goes down.   
I've also got tp-link deco wifi6 mesh routers.. I see that mentioned above, maybe something with tp-link? Tp-link doesnt have any real customization in the deco app.

**Comment by talormanda** [comment]
> I still have the issue with the ON_DEMAND setting as well. If I do 2-3 cameras, its fine, but if I go beyond that, everything goes down. I've also got tp-link deco wifi6 mesh routers.. I see that mentioned above, maybe something with tp-link? Tp-link doesnt have any real customization in the deco app.

Agreed. I can pull up the feeds one-by-one with the WebRTC/HLS/RTMP/RTSP link, but, using the webpage to view the static images is just junk and doesn't work. I have over 20 cameras.

**Comment by James62370** [comment]
I should clarify that with the ON_DEMAND setting, I'm able to use the Wyze App along with this wyze-bridge setup.  In the past, whenever I tried this, many of my Wyze devices would start dropping off making the Wyze App itself unuseable.  For me, the ON_DEMAND setting has greatly helped with that.  But as user jfarmer08 above mentions, I too am only able to get about 3 to 4 cameras working at the same time.  I believe this maybe a limitation with the Wyze Servers since even within their app, you cannot stream more than 4 cameras simultaneously.  Someone please correct me if I'm wrong.  However, I'm still glad to even get a few rtsp streams without needing to modify the firmware on the cameras.

**Comment by talormanda** [comment]
> I should clarify that with the ON_DEMAND setting, I'm able to use the Wyze App along with this wyze-bridge setup. In the past, whenever I tried this, many of my Wyze devices would start dropping off making the Wyze App itself unuseable. For me, the ON_DEMAND setting has greatly helped with that. But as user jfarmer08 above mentions, I too am only able to get about 3 to 4 cameras working at the same time. I believe this maybe a limitation with the Wyze Servers since even within their app, you cannot stream more than 4 cameras simultaneously. Someone please correct me if I'm wrong. However, I'm still glad to even get a few rtsp streams without needing to modify the firmware on the cameras.

This makes sense. You can't really get more than 4 working in landscape, so they might have implemented a limitation to make sure of that.

**Comment by mrlt8** [comment]
This is probably a UDP/Multicast issue. 

Check your networking gear for anything related with IGMP Snooping/IGMP Proxy/IGMP Version or IPTV related settings.

**Comment by talormanda** [comment]
> This is probably a UDP/Multicast issue.
> 
> Check your networking gear for anything related with IGMP Snooping/IGMP Proxy/IGMP Version or IPTV related settings.

Check it and do what? Enable it? Disable it? What is recommended?

**Comment by mrlt8** [comment]
Enabling should attempt to prevent the packets from getting forwarded everywhere.

**Comment by talormanda** [comment]
> Enabling should attempt to prevent the packets from getting forwarded everywhere.

I will test enabling it.
